lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1770,6 +1772,7 @@ bool TypeChecker::typeCheckFunctionBodyUntil(FuncDecl *FD,
                                /*skipClosures*/false);
           if (E != FD->getSingleExpressionBody())
             FD->setSingleExpressionBody(E);
+          setAutoClosureDiscriminators(FD, BS);
```

Comment:
Are there test cases that would fail without this?

Reply:
Yup!  Added here: 
https://github.com/apple/swift/pull/23251/commits/7fb0a208b2d5dc6dcdade1e0ff2778502361371e